pkg/virt-handler/vm.go

```diff
@@ -524,8 +524,12 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 					ifc := v1.VirtualMachineInstanceNetworkInterface{
 						Name: network.Name,
 						IP:   podIface.PodIP,
```

Comment:
If we're going to have multiple PodIPs in `IPs`, then this might also need to be changed according to the doc change here #3241   
@phoracek your view?

Reply:
do you mean to change the doc ? or ?

Reply:
It should match the Kubernetes API for PodIPs:

```
    // podIPs holds the IP addresses allocated to the pod. If this field is specified, the 0th entry must
    // match the podIP field. Pods may be allocated at most 1 value for each of IPv4 and IPv6. This list
    // is empty if no IPs have been allocated yet.
```

https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3555

I believe that Or's change matches that, but it is not clear from it. Maybe we should pick the first one explicitly here instead of depending on Kubernetes. I'm not sure which approach is better though.

Or did you mean something else?

Reply:
0th entry do match podIP, and i always put there IPv4 (as it was before my change, and as its on k8s)
more over dockershim says there isnt yet even a term "primary" on dual stack.
we use `addrList, err := Handler.AddrList(link, netlink.FAMILY_ALL)` and not depend on k8s order (maybe in the future we should), but put now IPv4 always on 0th entry and on PodIP (if IPv4 exists)
is it fine ?

Reply:
> and i always put there IPv4 if exists

The fact that the only valid primary IP is currently IPv4 is a side effect of a bug. We should follow the API specification and expect that Kubernetes reports the right first address in the list.

Is there any reason not to?

Reply:
it contradicts what you wrote before "Maybe we should pick the first one explicitly here instead of depending on Kubernetes", isnt it ?

but i can try to if needed, i assume the virt-launcher already have spec where i can read from and so on at that point, which what i would need.

just imo maybe we should focus first on what k8s support now, because it doesnt going to change soon, and we still have some important milestones, but np, whatever you suggest.

Reply:
> it contradicts what you wrote before "Maybe we should pick the first one explicitly here instead of depending on Kubernetes", isnt it ?

Both are following the API specification. What is contradicting?

> but i can try to if needed, i assume the virt-launcher already have spec where i can read from and so on at that point, which what i would need.

Read what?

> just imo maybe we should focus first on what k8s support now, because it doesnt going to change soon, and we still have some important milestones, but np, whatever you suggest.

What of the suggested is not "Kubernetes support"?

Reply:
> > it contradicts what you wrote before "Maybe we should pick the first one explicitly here instead of depending on Kubernetes", isnt it ?
> 
> Both are following the API specification. What is contradicting?

From what i understood, sorry if i was wrong, the quote sentence above says dont be depend on k8s and its order but use one of them explicitly always.

> 
> > but i can try to if needed, i assume the virt-launcher already have spec where i can read from and so on at that point, which what i would need.
> 
> Read what?

I should fetch (=read) the scheme and check what ip is in the PodIP and put the same in the VMI from what i understood now you suggest.

> 
> > just imo maybe we should focus first on what k8s support now, because it doesnt going to change soon, and we still have some important milestones, but np, whatever you suggest.
> 
> What of the suggested is not "Kubernetes support"?

K8s only support IPv4 as the first field on PodIPs and the one of PodIP in case of dual stack, so we are creating the future before its supported, but also before we support dual stack.

But np , whatever you suggest, i will be on it.
Thank

Reply:
done , in a separate commit that will be squashed at the end of the review.